### PR TITLE
Dockerfile: update to go 1.20

### PR DESCRIPTION
**Pull request by crazy-max**
Also update alpine to 3.17.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,17 +1,16 @@
 # syntax=docker/dockerfile:1
 
-ARG GO_VERSION=1.18
-
-ARG ALPINE_VERSION=3.16
-ARG ALPINE_BASE=alpine:${ALPINE_VERSION}
+ARG GO_VERSION=1.19
```
Comment (tonistiigi) [inline Dockerfile:3]: Not 1.20?

File: hack/dockerfiles/lint.Dockerfile [hunk 2]
```diff
@@ -1,8 +1,11 @@
 # syntax=docker/dockerfile:1
 
-FROM golang:1.18-alpine
+ARG GO_VERSION=1.19
+ARG ALPINE_VERSION=3.17
+
+FROM golang:${GO_VERSION}-alpine${ALPINE_VERSION}
```
Comment (tonistiigi) [inline hack/dockerfiles/lint.Dockerfile:6]: Is there a point in configuring alpine version here? The alpine part of these stages is not really used and probably only certain configurations are available anyway.

File: Dockerfile (hunk 1 above)
Comment (crazy-max) [reply Dockerfile]: Ah yes it was not GA when I made this changes last year. Updated to 1.20.

File: hack/dockerfiles/lint.Dockerfile (hunk 2 above)
Comment (crazy-max) [reply hack/dockerfiles/lint.Dockerfile]: Actually no. Removed.

**Review by tonistiigi — APPROVED** [review]

**Pull request merged**